db/seeds/source_types.yml

```diff
@@ -123,6 +123,32 @@ ansible-tower:
         :condition:
           :when: endpoint.verify_ssl
           :is: true
+google:
+  :product_name: Google Cloud Engine
+  :vendor: Google
+  :icon_url: "/apps/chrome/assets/images/partners-icons/google-cloud-engine.svg" # TODO
+  :schema:
+    :authentication:
+    - :type: username_password
+      :name: Project ID and Service Account JSON
+      :fields:
+      - :component: text-field
+        :name: authentication.authtype
+        :hideField: true
+        :initializeOnMount: true
+        :initialValue: username_password
+        :label: Project ID
+      - :component: textarea
+        :name: authentication.password
+        :label: Service Account JSON
+    :endpoint:
```

Comment:
Would we be able to leverage the new `Application` `extra` field to store the `dataset`?

Something like this
```
      - :name: application.extra.dataset
        :stepKey: cost-management
        :component: text-field
        :label: Dataset name
        :isRequired: true
```

Reply:
Yes, I agree! 👍  We will do it anyway in the future and doing it now will prevent any painful migration.

(You should also provide a validate in that object:

```
        :validate:
        - :type: required
```

`isRequired` is just a boolean to show the required marker.)

Reply:
added